IntegrationTests/tests_04_performance/test_01_resources/template/HookedFunctions/Sources/HookedFunctions/src/hooked-functions.c

```diff
@@ -25,30 +27,50 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+/* a big block of memory that we'll use for recursive mallocs */
+static char g_recursive_malloc_mem[10 * 1024 * 1024] = {0};
+/* the index of the first free byte */
+static _Atomic ptrdiff_t g_recursive_malloc_next_free_ptr = ATOMIC_VAR_INIT(0);
+
 #define DYLD_INTERPOSE(_replacement,_replacee) \
    __attribute__((used)) static struct { const void *replacement; const void *replacee; } _interpose_##_replacee \
             __attribute__ ((section("__DATA,__interpose"))) = { (const void *)(unsigned long)&_replacement, (const void *)(unsigned long)&_replacee };
 #define LIBC_SYMBOL(_fun) "" # _fun
 
-void replacement_free(void *ptr) {
-    if (ptr) {
-        inc_free_counter();
+static __thread bool g_in_malloc = false;
+static __thread bool g_in_realloc = false;
+static __thread bool g_in_free = false;
+static __thread void *(*g_libc_malloc)(size_t) = NULL;
+static __thread void *(*g_libc_realloc)(void *, size_t) = NULL;
+static __thread void (*g_libc_free)(void *) = NULL;
```

Comment:
Is this number of pointers right?

Reply:
@Lukasa I can't quite parse that :)

Reply:
Should this be `static __thread void *(*g_libc_free)(void *) = NULL;`?

Reply:
@Lukasa no, `free` doesn't return anything (your signature would be a `void *` return).

Reply:
Oh, yes, important to read the declaration properly instead of just counting asterisks.